src/engraving/dom/edit.cpp

```diff
@@ -2179,7 +2179,10 @@ void Score::cmdFlip()
     };
 
     for (EngravingItem* e : el) {
-        if (e->isNote() || e->isStem() || e->isHook()) {
+        if (e->hasVoiceApplicationProperties()) {
+            PlacementV curPlacement = e->getProperty(Pid::PLACEMENT).value<PlacementV>();
+            e->undoChangeProperty(Pid::DIRECTION, curPlacement == PlacementV::ABOVE ? DirectionV::DOWN : DirectionV::UP);
```

Comment:
Is it intended that this code is called _and_ the code below in `else if (e->isStaffText() || … || e->isDynamic()`?
And do we need to use this `flipOnce` thing to prevent unexpectedly flipping the same element twice?

Reply:
(I don't know what `flipOnce` does or means, but given that this works as expected it's probably not needed)

Reply:
I believe it mainly has to do with spanner segments. When a spanner is broken into two segments, and for each of these segments you'd flip the spanner, you have flipped the spanner twice. When using `flipOnce`, the spanner gets stored in `alreadyFlippedElements`, so that is handled only once.